packages/zendesk-adapter/src/filters/support_address.ts

```diff
@@ -0,0 +1,142 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  Change,
+  Element,
+  getChangeData,
+  InstanceElement,
+  isAdditionOrModificationChange,
+  isInstanceChange,
+  isInstanceElement,
+  isReferenceExpression, isTemplateExpression,
+  ReferenceExpression,
+  TemplateExpression,
+  TemplatePart,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { extractTemplate } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import { FilterCreator } from '../filter'
+import { BRAND_TYPE_NAME, SUPPORT_ADDRESS_TYPE_NAME } from '../constants'
+
+const log = logger(module)
+
+
+const referenceEmail = ({ emailPart, brandInstances }: {
+  emailPart: string
+  brandInstances: Record<string, InstanceElement>
+}): TemplatePart[] => {
+  // emailPart should be of the form {username}@{subdomain}.{domain} (usually zendesk.com)
+  // zendesk subdomain cannot have a dot (.) in it.
+  const splitedEmail = emailPart.split(/^([^@]+)@([^.]+)\.(.+)$/).filter(v => !_.isEmpty(v))
+  if (splitedEmail.length !== 3) {
+    return [emailPart]
+  }
+  const [username, subdomain, domain] = splitedEmail
+  const elem = brandInstances[subdomain]
+  if (elem) {
```

Comment:
nit: we try to be explicit in boolean checks
```suggestion
  if (elem !== undefined) {
```